docs/how-to-write-function.md

```diff
@@ -4,104 +4,150 @@ This document sets out some guidelines for writing functions for Scalar DL.
+
+`contractArgument` and `contractProperties` are the corresponding contract's argument and properties. See [How to Write a Contract](how-to-write-contract.md) to understand what they are.
 
 ### How to use Functions
 
-The Function feature is enabled by default; thus, nothing needs to be configured in Ledger except for the following things.
-If you want to disable the feature, please set `scalar.dl.ledger.function.enabled` to `false` in the properties of Ledger.
+The Function feature is enabled by default; thus, nothing needs to be configured in Ledger except for the following things. If you want to disable the feature, please set `scalar.dl.ledger.function.enabled` to `false` in the properties of Ledger.
 
 #### Add an application-specific schema
 
-Since Functions can read and write arbitrary records through the Scalar DB CRUD interface, Scalar DL can't define the database schema for the Function by itself.
-It is the applications' owner's responsibility to define such schema and apply it to the database by themselves or asking system admins to do it depending on who owns and manages the database.
+Since Functions can read and write arbitrary records through the Scalar DB CRUD interface, Scalar DL can't define the database schema for the Function by itself. It is the applications' owner's responsibility to define such schema and apply it to the database by themselves or asking system admins to do it depending on who owns and manages the database.
 
 #### Register a Function
 
 You then need to register a Function to Ledger before used like you register a Contract.
 
 ```
-register-function --properties client.properties --function-id test-function --function-binary-name com.example.function.TestFunction --function-class-file /path/to/TestFunction.class
+client/bin/register-function --properties client.properties --function-id test-function --function-binary-name com.example.function.TestFunction --function-class-file /path/to/TestFunction.class
 ```
 
-#### Execute a registered Function
+#### Execute a Function
 
-Functions that are being executed are specified in a contract argument in a JSON format with `_functions_` as a key and an array of function IDs as a value as follows:
+You can specify a Function to execute along with a Contract to execute.
+For example, you can execute a function as follows with the command-line tool.
 
 ```
-execute-contract --properties client.properties --contract-id test-contract --contract-argument '{..., "_functions_": ["test-function"]}' --function-argument '{...}'
+client/bin/execute-contract --properties client.properties --contract-id test-contract --contract-argument '{...}' --function-id test-function --function-argument '{...}'
```

Comment:
This is a question.

In the previous way, I can execute several functions in one contract execution by passing `list` of function IDs in the Contract argument.
In the new way, can I execute several functions in one contract execution?
For example, as follows.
```console
client/bin/execute-contract --properties client.properties --contract-id test-contract --contract-argument '{...}' --function-id test-function-1 --function-id test-function-2 --function-id test-function-3 ... --function-argument '{...}' 
```
Or, do I need to use `invoke()` method in the Function (need to nest the Functions) to execute several functions?

Reply:
Thank you for the question.
It's a very good point.
The new interface doesn't allow users to specify multiple functions to make it consistent with the contract usage. 
So, if you want to execute multiple functions in one contract execution, you need to call them from one top-level function.
But, the old interface can still specify multiple functions, so Scalar DL still has the feature internally.

Reply:
I see. I understand.
Thank you!